apps/web/src/components/profile/account-deletion-form.tsx

```diff
@@ -0,0 +1,132 @@
+"use client";
+
+import { useTransition } from "react";
+import { deleteAccount } from "@/actions/profile";
+import { deleteAccountSchema } from "@/lib/schema/profile";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Loader2Icon } from "lucide-react";
+import { useForm } from "react-hook-form";
+import { signOutAction } from "@/actions/auth";
+import { useState } from "react";
+import type { z } from "zod";
+
+import {
+    Button,
+    Card,
+    CardContent,
+    CardDescription,
+    CardHeader,
+    CardTitle,
+    Dialog,
+    DialogContent,
+    DialogDescription,
+    DialogHeader,
+    DialogTitle,
+    Form,
+    FormControl,
+    FormField,
+    FormItem,
+    FormMessage,
+    Input,
+} from "@energyleaf/ui";
+import { useToast } from "@energyleaf/ui/hooks";
+
+interface Props {
+    id: string;
+}
+
+export default function AccountDeletionForm({ id }: Props) {
+    const [isPending, startTransition] = useTransition();
+    const [open, setOpen] = useState(false);
+    const { toast } = useToast();
+    const form = useForm<z.infer<typeof deleteAccountSchema>>({
+        resolver: zodResolver(deleteAccountSchema),
+        defaultValues: {
+            password: "",
+        },
+    });
+
+    function onSubmit(data: z.infer<typeof deleteAccountSchema>) {
+        startTransition(async () => {
+            setOpen(false);
+            try {
+                await deleteAccount(data, id);
+                toast({
+                    title: "Erfolgreich gelöscht",
+                    description: "Dein Account wurde erfolgreich gelöscht",
+                });
+                await signOutAction();
+            } catch (e) {
+                if (e.message === "Passwords do not match") {
```

Comment:
Das ist meiner Meinung nach nicht so schön. Wenn die Message in deleteAccount geändert wird (z. B. wenn später mal Mehrsprachigkeit umgesetzt wird), greift die Logik hier nicht und es ist auch nicht direkt ersichtlich, dass man das hier auch noch anpassen müsste.
Schöner wäre es aus meiner Sicht, einen neuen Error-Typen anzulegen, auf den man dann hier prüft.

Reply:
Wie @Big0x44 schon sagt wir sollten uns hier einheitliche Errors überlegen.

```typescript
throw new Error("password/mismatch")
```

Reply:
> Wie @Big0x44 schon sagt wir sollten uns hier einheitliche Errors überlegen.
> 
> ```ts
> throw new Error("password/mismatch")
> ```

Ist "password/mismatch" nicht einfach nur die Error Message? Ich hätte jetzt folgendes gemacht:

```ts
export class PasswordsDoNotMatchError extends Error {
    constructor() {
      super();
      Object.setPrototypeOf(this, PasswordsDoNotMatchError.prototype)
    }
}
...
throw new PasswordsDoNotMatchError()
```

Reply:
Ich persönlich finde den Ansatz mit PasswordsDoNotMatchError besser, weil man relativ easy die Verweise suchen kann und auch der Build gar nicht erst durchläuft, wenn man den Error umbenennen sollte